src/bin/grin.rs

```diff
@@ -357,8 +350,26 @@ fn wallet_command(wallet_args: &ArgMatches) {
 		wallet_config.check_node_api_http_addr = sa.to_string().clone();
 	}
 
-	match wallet_args.subcommand() {
+	// Derive the keychain based on seed from seed file and specified passphrase.
+	// Generate the initial wallet seed if we are running "wallet init".
+	let keychain = match wallet_args.subcommand() {
+		("init", Some(_)) => {
+			let wallet_seed = wallet::WalletSeed::init_file(&wallet_config)
+				.expect("Failed to init wallet seed file.");
+			wallet_seed.derive_keychain(&"")
+				.expect("Failed to derive keychain from seed file and passphrase.")
```

Comment:
Maybe just return here? Someone may be tempted to add code underneath without thinking twice about it that wouldn't be applicable to init.

Reply:
Oh right... I was wondering how to stop here, never thought to just `return`...